Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4734,12 +4734,6 @@ void WebPageProxy::preferencesDidChange()
 
 void WebPageProxy::didCreateMainFrame(FrameIdentifier frameID)
 {
-    // The DecidePolicyForNavigationActionSync IPC is synchronous and may therefore get processed before the DidCreateMainFrame one.
-    // When this happens, decidePolicyForNavigationActionSync() calls didCreateMainFrame() and we need to ignore the DidCreateMainFrame
-    // IPC when it later gets processed.
-    if (m_mainFrame && m_mainFrame->frameID() == frameID)
```

Comment:
We should replace with an assertion to make sure we were successful.

Reply:
There's already a MESSAGE_CHECK(m_process, !m_mainFrame); below.